packages/itwin/tree-widget/src/e2e-tests/TreeWidget.test.ts

```diff
@@ -19,108 +20,137 @@ test.beforeEach(async ({ page, baseURL }) => {
+
+    test("node with active filtering - information message", async ({ page }) => {
+      const physicalModelNode = locateNode(treeWidget, "ProcessPhysicalModel");
+
+      // hover the node for the button to appear
+      await physicalModelNode.hover();
+      await physicalModelNode.getByTitle("Apply filter").click();
+
+      await locateInstanceFilter(page).waitFor();
+      await selectPropertyInDialog(page, "Is Private");
+
+      await page.getByRole("button", { name: "Apply" }).click();
+
+      // hover the node for the button to appear
+      await physicalModelNode.hover();
+      await treeWidget.getByTitle("Clear active filter").waitFor();
+
+      await takeScreenshot(page, treeWidget);
+    });
+
+    test("node with active filtering - filtered nodes", async ({ page }) => {
+      const physicalModelNode = locateNode(treeWidget, "ProcessPhysicalModel");
+
+      // hover the node for the button to appear
+      await physicalModelNode.hover();
+      await physicalModelNode.getByTitle("Apply filter").click();
+
+      await locateInstanceFilter(page).waitFor();
+      await selectPropertyInDialog(page, "Is Private");
+
+      await page
+        .getByRole("combobox")
+        .filter({ has: page.getByText("Is true") })
+        .click();
+      await page.getByRole("option", { name: "Is true" }).waitFor();
+      await page.getByRole("option", { name: "Is false" }).click();
+
+      await page.getByRole("button", { name: "Apply" }).click();
+
+      // hover the node for buttons to appear
+      await physicalModelNode.hover();
+      await treeWidget.getByTitle("Clear active filter").waitFor();
+
+      // wait for node at the bottom to be visible/loaded
+      await locateNode(treeWidget, lastNodeLabel).waitFor();
+      await takeScreenshot(page, treeWidget);
+    });
+
+    test("selected node", async ({ page }) => {
+      const node = locateNode(treeWidget, "BayTown");
+      await node.click();
+
+      // wait for node to become selected
+      await expect(node).toHaveClass(/is-selected/);
+      await takeScreenshot(page, treeWidget);
+    });
+
+    test("search", async ({ page }) => {
+      await treeWidget.getByText("BayTown").waitFor();
+      await treeWidget.getByTitle("Search for something").click();
+      await treeWidget.getByPlaceholder("Search...").fill("Model");
+      await treeWidget.locator(".components-activehighlight").waitFor();
+      await takeScreenshot(page, treeWidget);
+    });
+
+    test("button dropdown", async ({ page }) => {
+      await treeWidget.getByTitle("Search for something").click();
+      await treeWidget.getByTitle("More").click();
+      await page.locator(".tree-header-button-dropdown-container").waitFor();
+      await takeScreenshot(page, treeWidget);
+    });
+
+    test("tree selector", async ({ page }) => {
+      await treeWidget.getByRole("combobox").click();
+      await page.getByRole("listbox").waitFor();
+      await takeScreenshot(page, treeWidget);
+    });
```

Comment:
I ran e2e tests locally and it looks like these tests might be flaky depending on how quickly roots nodes are loaded. I would suggest adding `waitFor` to check if root node is visible before doing interactions and taking screenshot.